.travis.yml

```diff
@@ -1,83 +1,52 @@
+    - PHP_VERSION=5.6 VARNISH_VERSION=4.1 VARNISH_MODULES_VERSION=0.10.2 COMPOSER_FLAGS="--prefer-lowest"
+    - PHP_VERSION=7.0 VARNISH_VERSION=5.0
+    - PHP_VERSION=7.1 DOCCHECK=true DEPENDENCIES="symfony/lts:^2"
+    - PHP_VERSION=7.2
+    #- PHP_VERSION=7.2 STABILITY="dev"
 
 branches:
   only:
     - master
     # Build maintenance branches for older releases if needed. such branches should be named like "1.2"
     - '/^\d+\.\d+$/'
 
-before_install:
-    - if [[ $COVERAGE != true ]]; then phpenv config-rm xdebug.ini || true; fi
-    - if ! [ -z "$STABILITY" ]; then composer config minimum-stability ${STABILITY}; fi;
-    - if ! [ -v "$DEPENDENCIES" ]; then composer require --no-update ${DEPENDENCIES}; fi;
+# For local dev, remember to append --build if you have done changes to docker config or want to change php/varnish versions.
+install: docker-compose up -d
 
-install:
-  - composer update $COMPOSER_FLAGS --prefer-dist --no-interaction
 
 before_script:
-  # Install Varnish
-  - |
-    set -e
-    curl -L https://packagecloud.io/varnishcache/varnish${VARNISH_VERSION//./}/gpgkey | sudo apt-key add -
-    curl -L "https://packagecloud.io/install/repositories/varnishcache/varnish${VARNISH_VERSION//./}/config_file.list?os=ubuntu&dist=trusty&source=script" | sudo tee -a /etc/apt/sources.list
-    cat /etc/apt/sources.list
-    sudo apt-get update -qq
-    sudo apt-get install -qq varnish
-  - if [ "$DOCCHECK" = true ]; then sudo apt-get install -qq python-sphinx enchant; fi
-  - if [ "$DOCCHECK" = true ]; then sudo pip install -r doc/requirements.txt; fi
-  # Install NGINX
-  - sh ./tests/install-nginx.sh
+  - if ! [ -z "$STABILITY" ]; then docker-compose exec fos sh -c "composer config minimum-stability ${STABILITY}"; fi;
+  - if ! [ -v "$DEPENDENCIES" ]; then docker-compose exec fos sh -c "composer require --no-update ${DEPENDENCIES}"; fi;
+  - docker-compose exec fos sh -c "composer update $COMPOSER_FLAGS --prefer-dist --no-interaction"
+  #- if [ "$DOCCHECK" = true ]; then sudo pip install -r doc/requirements.txt; fi
 
 script:
-  - composer validate --strict --no-check-lock
-  - vendor/bin/phpunit $PHPUNIT_FLAGS
-  - if [[ "$DOCCHECK" = true ]]; then make -C doc SPHINXOPTS='-nW' html; fi
-  - if [[ "$DOCCHECK" = true ]]; then make -C doc spelling; fi
+  - docker-compose exec fos sh -c "composer validate --strict --no-check-lock"
+  - docker-compose exec fos sh -c "vendor/bin/phpunit $PHPUNIT_FLAGS"
+  #- if [[ "$DOCCHECK" = true ]]; then make -C doc SPHINXOPTS='-nW' html; fi
+  #- if [[ "$DOCCHECK" = true ]]; then make -C doc spelling; fi
 
-after_script:
-  - wget https://scrutinizer-ci.com/ocular.phar && php ocular.phar code-coverage:upload --format=php-clover coverage.clover
+#after_script:
+  # Todo fix to use curl in image, and either install xdebug OR change to use phpdbg, then re add $PHPUNIT_FLAGS usage
+  #- wget https://scrutinizer-ci.com/ocular.phar && php ocular.phar code-coverage:upload --format=php-clover coverage.clover
```

Comment:
i can live with dropping scrutinizer from our test suite. maybe we could introduce phpstan instead (in a separate pull request, though)

Reply:
Either way we need to make sure it can be done inside the image. If this can be done in followup then I can try to wrap this up soonish, like end of week.